File: js/leetcode/1-two-sums.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  hash = {}

  for (let i = 0; i < nums.length; i++) {
    if (target - nums[i] in hash) {
      return [i, hash[target - nums[i]]];
    }
    hash[nums[i]] = i;
  }
};

File: c#/leetcode/1-two-sums.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        
        for (int i=0; i<nums.Length; i++) {
            
            if (dict.ContainsKey(target-nums[i])) {
                return new int[]{dict[target-nums[i]], i};
            }
            else if (!dict.ContainsKey(nums[i])) {
                dict.Add(nums[i], i);
            }
        }
        return new int[]{-1,-1};
    }
}